### PR TITLE
Add intellisence docs

### PR DESCRIPTION
**Pull request by malinthar**
Add VS Code docs for intelliSence support.
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/39060

### CODE REVIEW

File: en/docs/edit-the-code/intellisense.md [hunk 1]
```diff
@@ -1 +1,44 @@
-# IntelliSense
\ No newline at end of file
+# IntelliSense
+
+### Code Completion & Snippets
```
Comment (anupama-pathirage) [inline en/docs/edit-the-code/intellisense.md:3]:
```suggestion
### Code completion and snippets
```

File: en/docs/edit-the-code/intellisense.md [hunk 2]
```diff
@@ -1 +1,44 @@
...
+
+<img src="https://github.com/wso2/docs-ballerina-visual-studio-code-plugin/blob/main/en/docs/img/type-guard-completion.gif?raw=true" width="100%" />
+
+#### Filling required fields of a record
+The `Fill Record required fileds` completion item is provided for filling the remaining fields of a `Record` typed value.
+
+<img src="https://github.com/wso2/docs-ballerina-visual-studio-code-plugin/blob/main/en/docs/img/fill-req-fileds-completion.gif?raw=true" width="100%" />
+
+
+### Help via Hover
```
Comment (anupama-pathirage) [inline en/docs/edit-the-code/intellisense.md:36]:
```suggestion
### Help via hover
```

File: en/docs/edit-the-code/intellisense.md [hunk 3]
```diff
@@ -1 +1,44 @@
...
+When hovering over a symbol name, you will be provided with quick information about the particular symbol. For example, when hovering over a function name, you will be prompted with the associated documentation.
+
+<img src="https://github.com/wso2/docs-ballerina-visual-studio-code-plugin/blob/main/en/docs/img/symbol-information-on-hover.gif?raw=true" width="100%" />
+
+### Signature Help
```
Comment (anupama-pathirage) [inline en/docs/edit-the-code/intellisense.md:41]:
```suggestion
### Signature help
```

File: en/docs/edit-the-code/intellisense.md [hunk 4]
```diff
@@ -1 +1,44 @@
-# IntelliSense
\ No newline at end of file
+# IntelliSense
+
+### Code Completion & Snippets
+The extension provides you with suggestions on variables, keywords, and code snippets of language constructs. (such as functions, type definitions, services, iterable constructs, etc.)
+
+<img src="https://github.com/wso2/docs-ballerina-visual-studio-code-plugin/blob/main/en/docs/img/code-completion.gif?raw=true" width="100%" />
+
+Following are some key features of the code completion support provided by the plugin.
```
Comment (anupama-pathirage) [inline en/docs/edit-the-code/intellisense.md:8]: The word `plugin` and `extension` are used interchangeably. Stick to one through out the content.

File: en/docs/edit-the-code/intellisense.md [hunk 5]
```diff
@@ -1 +1,44 @@
-# IntelliSense
\ No newline at end of file
+# IntelliSense
+
+### Code Completion & Snippets
+The extension provides you with suggestions on variables, keywords, and code snippets of language constructs. (such as functions, type definitions, services, iterable constructs, etc.)
```
Comment (anupama-pathirage) [inline en/docs/edit-the-code/intellisense.md:4]:
```suggestion
The extension provides suggestions on variables, keywords, and code snippets of language constructs. (such as functions, type definitions, services, iterable constructs, etc.)
```

File: en/docs/edit-the-code/intellisense.md [hunk 6]
```diff
@@ -1 +1,44 @@
...
+
+<img src="https://github.com/wso2/docs-ballerina-visual-studio-code-plugin/blob/main/en/docs/img/code-completion.gif?raw=true" width="100%" />
+
+Following are some key features of the code completion support provided by the plugin. 
+
+#### Context aware completion items
```
Comment (anupama-pathirage) [inline en/docs/edit-the-code/intellisense.md:10]:
```suggestion
#### Context-aware completion items
```

File: en/docs/edit-the-code/intellisense.md [hunk 7]
```diff
@@ -1 +1,44 @@
...
+The completion item list is sorted based on the context of the current cursor position.
+
+<img src="https://github.com/wso2/docs-ballerina-visual-studio-code-plugin/blob/main/en/docs/img/context-aware-completions.gif?raw=true" width="100%" />
+
+#### Suggestions for service templates
+Service templates corresponding to each available listener is provided in the list of completion items.
```
Comment (anupama-pathirage) [inline en/docs/edit-the-code/intellisense.md:16]:
```suggestion
Service templates corresponding to each available listener are provided in the list of completion items.
```

File: en/docs/edit-the-code/intellisense.md [hunk 8]
```diff
@@ -1 +1,44 @@
...
+
+<img src="https://github.com/wso2/docs-ballerina-visual-studio-code-plugin/blob/main/en/docs/img/service-template-completion.gif?raw=true" width="100%" />
+
+#### Completion support for iterables
+The `foreach` and `foreach i` completion items are provided for iterable variables.
+
+<img src="https://github.com/wso2/docs-ballerina-visual-studio-code-plugin/blob/main/en/docs/img/foreach-completion.gif?raw=true" width="100%" />
+
+#### Completion support for type guarding a variable
+The `typeguard` completion item is provided for union typed variables.
```
Comment (anupama-pathirage) [inline en/docs/edit-the-code/intellisense.md:26]:
```suggestion
The `typeguard` completion item is provided for union-typed variables.
```

File: en/docs/edit-the-code/intellisense.md [hunk 9]
```diff
@@ -1 +1,44 @@
...
+
+<img src="https://github.com/wso2/docs-ballerina-visual-studio-code-plugin/blob/main/en/docs/img/type-guard-completion.gif?raw=true" width="100%" />
+
+#### Filling required fields of a record
+The `Fill Record required fileds` completion item is provided for filling the remaining fields of a `Record` typed value.
```
Comment (anupama-pathirage) [inline en/docs/edit-the-code/intellisense.md:31]:
```suggestion
The `Fill Record required fields` completion item is provided for filling the remaining fields of a `Record` typed value.
```

File: en/docs/edit-the-code/intellisense.md [hunk 10]
```diff
@@ -1 +1,44 @@
...
+The extension provides you with suggestions on variables, keywords, and code snippets of language constructs. (such as functions, type definitions, services, iterable constructs, etc.)
+
+<img src="https://github.com/wso2/docs-ballerina-visual-studio-code-plugin/blob/main/en/docs/img/code-completion.gif?raw=true" width="100%" />
+
+Following are some key features of the code completion support provided by the plugin. 
+
+#### Context aware completion items
+The completion item list is sorted based on the context of the current cursor position.
+
+<img src="https://github.com/wso2/docs-ballerina-visual-studio-code-plugin/blob/main/en/docs/img/context-aware-completions.gif?raw=true" width="100%" />
```
Comment (anupama-pathirage) [inline en/docs/edit-the-code/intellisense.md:13]: The recording in the gif seems a bit slow. i.e., waiting too much time on the appeared menu. Can we fix that?

Comment (malinthar) [reply en/docs/edit-the-code/intellisense.md:13]: Done

**Review by anupama-pathirage — APPROVED** [review]

**Pull request merged**